core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -295,13 +300,13 @@ object LAFuture {
     }
   }
 
-  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit) {
+  private def executeWithObservers(scheduler: LAScheduler, f: () => Unit, wrapper:CommonLoanWrapper) {
     val cur = threadInfo.get()
     scheduler.execute(() => {
       val old = threadInfo.get()
       threadInfo.set(cur)
       try {
-        f()
+        wrapper(f())
```

Comment:
`Option` means we're doing a conditional check vs a function invocation, in what is likely to be the most common case for most users of the API.